front/app/containers/Admin/projects/project/analysis/Insights/Files/FileSelectionView/index.tsx

```diff
@@ -1,16 +1,18 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
```

Comment:
I have a couple suggestions for this file:
- Using lodash `isEqual`
- Using the files from the analysis `(analysis?.data.relationships.files?.data.map((file) => file.id)` directly when instantiating the `previousFileIdsRef` and `methods` so we can remove the useEffect (unless I've made a mistake and we do still need it).

I'll send the final code in a DM on Slack - might be good to test once more just to make sure I didn't miss anything though! :)

Reply:
Thanks! Your code was much better 🤩